### PR TITLE
feat(vertexai): pdf input

### PR DESCRIPTION
**Pull request by Deleplace**
Go version of the Python sample at https://github.com/GoogleCloudPlatform/python-docs-samples/blob/main/generative_ai/gemini_pdf_example.py

Region tag `generativeaionvertexai_gemini_pdf`

### CODE REVIEW

**Review by gericdong — APPROVED** [review]

File: vertexai/multimodal-pdf/pdf.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	model.SetTemperature(0.4)
+
+	part := genai.FileData{
+		MIMEType: "application/pdf",
+		FileURI:  pdf,
+	}
+
+	res, err := model.GenerateContent(ctx, part, genai.Text(prompt))
+	if err != nil {
+		return fmt.Errorf("unable to generate contents: %v", err)
```
Comment (grayside) [inline vertexai/multimodal-pdf/pdf.go:57]: blocking: This and other fmt.Errorf instances, please wrap errors with %w.

File: vertexai/multimodal-pdf/pdf.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/vertexai/genai"
+)
+
+// generateContentFromPDF generates a response into w, based upon the prompt
+// and the pdf file provided.
+// pdf is a Google Cloud Storage path starting with "gs://"
+func generateContentFromPDF(w io.Writer, prompt, pdf, projectID, location, modelName string) error {
```
Comment (grayside) [inline vertexai/multimodal-pdf/pdf.go:31]: suggestion: Follow the decision made as part of https://github.com/GoogleCloudPlatform/golang-samples/pull/4110#pullrequestreview-2022822492

File: vertexai/multimodal-pdf/pdf.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	"errors"
+	"fmt"
+	"io"
+
+	"cloud.google.com/go/vertexai/genai"
+)
+
+// generateContentFromPDF generates a response into w, based upon the prompt
+// and the pdf file provided.
+// pdf is a Google Cloud Storage path starting with "gs://"
```
Comment (grayside) [inline vertexai/multimodal-pdf/pdf.go:30]: suggestion: Similar to https://github.com/GoogleCloudPlatform/golang-samples/pull/4113#discussion_r1579782136, suggest making the purpose of the code vs. what it does a little more clear.

Secondarily, my preference is to make the top-line comment independent of reading the function signature, so I would use "provided io.Writer" over specifying the variable name `w`.

File: vertexai/multimodal-pdf/pdf_test.go [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+
+	"github.com/GoogleCloudPlatform/golang-samples/internal/testutil"
+)
+
+func Test_generateContentFromPDF(t *testing.T) {
+	tc := testutil.SystemTest(t)
+
+	buf := new(bytes.Buffer)
+	prompt := `
+		Your are a very professional document summarization specialist.
```
Comment (grayside) [inline vertexai/multimodal-pdf/pdf_test.go:30]: suggestion:
```suggestion
	buf := new(bytes.Buffer)
	prompt := `
		You are a very professional document summarization specialist.
```

**Review by grayside — CHANGES_REQUESTED** [review]

File: vertexai/multimodal-pdf/pdf_test.go (hunk 4 above)
Comment (Deleplace) [reply vertexai/multimodal-pdf/pdf_test.go:30]: Good catch, thank you

File: vertexai/multimodal-pdf/pdf.go (hunk 1 above)
Comment (Deleplace) [reply vertexai/multimodal-pdf/pdf.go:57]: Awesome! I wasn't very familiar with `%w`, now I see it makes perfect sense

File: vertexai/multimodal-pdf/pdf_test.go (hunk 4 above)
Comment (Deleplace) [reply vertexai/multimodal-pdf/pdf_test.go]: Fixing upstream in Python: https://github.com/GoogleCloudPlatform/python-docs-samples/pull/11592

**Comment by Deleplace** [comment]
kokoro filing with `Test_functionCallsChat: rpc error: ....` but unrelated to the current code change

**Review by grayside — APPROVED** [review]

**Pull request merged**